gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdkExtension.java

```diff
@@ -24,11 +30,46 @@ public abstract class JdkExtension {
 
     public abstract Property<JdkDistributionName> getDistributionName();
 
+    private final Map<Os, JdkOsExtension> jdkOsExtensions = new HashMap<>();
```

Comment:
With `NamedDomain....` need you to give a string for the name (or implement `Named`, which bring in a needless gradle dep). I don't really think it buys much? Unless there's some magic you can do with it I'm not aware of.

Reply:
Given how it's used, I think it doesn't buy us as much, since we need it to statically exist. I believe it creates most of the methods for you, but that doesn't really matter if you're calling it from java